libs/wire-api/src/Wire/API/Routes/Version.hs

```diff
@@ -57,44 +57,63 @@ import Servant
 import Servant.Swagger
 import Wire.API.Routes.Named
 import Wire.API.VersionInfo
+import Wire.Arbitrary (Arbitrary, GenericUniform (GenericUniform))
 
 -- | Version of the public API.
 data Version = V0 | V1 | V2 | V3
-  deriving stock (Eq, Ord, Bounded, Enum, Show)
+  deriving stock (Eq, Ord, Bounded, Enum, Show, Generic)
   deriving (FromJSON, ToJSON) via (Schema Version)
+  deriving (Arbitrary) via (GenericUniform Version)
+
+versionString :: IsString a => Version -> a
+versionString V0 = "v0"
+versionString V1 = "v1"
+versionString V2 = "v2"
+versionString V3 = "v3"
+
+versionInt :: Integral i => Version -> i
+versionInt V0 = 0
+versionInt V1 = 1
+versionInt V2 = 2
+versionInt V3 = 3
 
 instance ToSchema Version where
-  schema =
-    enum @Integer "Version" . mconcat $
-      [ element 0 V0,
-        element 1 V1,
-        element 2 V2,
-        element 3 V3
```

Comment:
IIRC we wanted it explicit because we'll soon start dropping support for older versions in code. But I'm no longer sure.

Reply:
this is still the case, we're just doing the explicit enumeration in `versionString`, `versionInt` above now.